test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java

```diff
@@ -61,4 +76,19 @@ default void inject(Object testInstance) {
     default int order() {
         return 0;
     }
+
+    /**
+     * Provides methods to handle the common injection cases. See
+     * {@link QuarkusTestResourceLifecycleManager#inject(TestInjector)}
+     */
+    interface TestInjector {
+
+        /**
+         * @param fieldValue The actual value to inject into a test field
+         * @param fieldType The field's required type
+         * @param annotations Optional annotations that the field must have. If more than one are specified, the field
+         *        must be annotated with all the provided annotations
+         */
+        void injectIntoFields(Object fieldValue, Class<?> fieldType, Class<? extends Annotation>... annotations);
```

Comment:
Maybe you could add a variant with `javax.enterprise.util.TypeLiteral<?> fieldType` to handle the parameterized types but the problem is that you would have to specify the rules for matching... or rely on existing rules, e.g. from CDI. But these need to be inherently complex and thus it's probably not worth the effort.

Reply:
Good point!

I would prefer we start with this simple approach for now and keep your suggestion in mind for when / if people need something more complex